extensions/interactions/base_test.py

```diff
@@ -343,109 +344,230 @@ def test_default_interactions_are_valid(self):
-            short_response_directive_ts_file_content = utils.get_file_contents(
-                short_response_directive_ts_file)
-            ts_file_content = utils.get_file_contents(ts_file)
-            rules_service_ts_file_content = utils.get_file_contents(
-                rules_service_ts_file)
-            validation_service_ts_file_content = utils.get_file_contents(
-                validation_service_ts_file)
-
-            self.assertIn('oppiaInteractive%s' % interaction_id,
-                          interaction_directive_ts_file_content)
-            self.assertIn(
-                'oppiaResponse%s' % interaction_id,
-                response_directive_ts_file_content)
-            self.assertIn(
-                'oppiaShortResponse%s' % interaction_id,
-                short_response_directive_ts_file_content)
-            self.assertIn(
-                '%sRulesService' % (
-                    interaction_id[0] + interaction_id[1:]),
-                rules_service_ts_file_content)
-            self.assertIn(
-                '%sValidationService' % interaction_id,
-                validation_service_ts_file_content)
-
-            # Check that the html template includes js script for the
-            # interaction.
-            self.assertIn(
-                'oppia-interactive-%s.directive.ts' % hyphenated_interaction_id,
-                ts_file_content)
-            self.assertIn(
-                'oppia-response-%s.directive.ts' % hyphenated_interaction_id,
-                ts_file_content)
-            self.assertIn(
-                'oppia-short-response-%s.directive.ts'
-                % hyphenated_interaction_id, ts_file_content)
-            self.assertIn(
-                '%s-rules.service.ts' % hyphenated_interaction_id,
-                ts_file_content)
-            self.assertIn(
-                '%s-validation.service.ts' % hyphenated_interaction_id,
-                ts_file_content)
-
-            self.assertNotIn('<script>', interaction_directive_ts_file_content)
-            self.assertNotIn('</script>', interaction_directive_ts_file_content)
-            self.assertNotIn('<script>', response_directive_ts_file_content)
-            self.assertNotIn('</script>', response_directive_ts_file_content)
-            self.assertNotIn(
-                '<script>', short_response_directive_ts_file_content)
-            self.assertNotIn(
-                '</script>', short_response_directive_ts_file_content)
-            self.assertNotIn('<script>', rules_service_ts_file_content)
-            self.assertNotIn('</script>', rules_service_ts_file_content)
-            self.assertNotIn('<script>', validation_service_ts_file_content)
-            self.assertNotIn('</script>', validation_service_ts_file_content)
+            if interaction_id in INTERACTIONS_THAT_USE_COMPONENTS:
+                interaction_component_ts_file = os.path.join(
+                    directives_dir, 'oppia-interactive-%s.component.ts' % (
```

Comment:
Instead of repeating the whole thing, isn't it possible to just declare these filename strings separately initially and reuse all the actual checks?

Reply:
Done!